src/esp/core/logging.cpp

```diff
@@ -59,9 +59,6 @@ LoggingLevel levelFromName(const Corrade::Containers::StringView name) {
 }
 
 namespace {
-#ifdef CORRADE_BUILD_MULTITHREADED
-CORRADE_THREAD_LOCAL
-#endif
```

Comment:
~~Quick thought -- have a thread_local mutable instance and expose a non-thread-local const pointer?~~ Eh, no, that won't prevent races either, this way the originating thread can still modify the state while other threads read it.

I suppose a way would be to make the `LoggingContext` completely immutable (without `processEnvString()` or `reinitializeFromEnv()`), so reparsing the environment would mean recreating the whole instance, which, due to the RAII nature of it, is rather unlikely to happen in a non-thread-safe way? To me that sounds like a good thing to do in any case, immutable means less accidents.

Reply:
My rationale for making things mutable and having `reinitializeFromEnv()` was for python.  Since the logging context is a module level global there, you have to set the environment variable before importing habitat_sim.  Not a huge limitation but it seemed nice to let users set that in code if they so desired and just re-parse the environment variable on simulator spin-up.

Reply:
Okay, since Python doesn't really have a concept of multithreading anyway, what about exposing `reinitializeFromEnv()` as a Python-only thing, which would be a lambda that internally recreates the `LoggingContext` instance?

Reply:
I like this idea.  I'll do that.